api/v1alpha1/statuscheck_types.go

```diff
@@ -0,0 +1,230 @@
+	// such as "300ms", "-1.5h" or "2h45m".
+	// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+	// +optional
+	Duration *string `json:"duration,omitempty" webhook:"Duration"`
+
+	// TimeoutSeconds defines the number of seconds after which
+	// an execution of status check times out.
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	TimeoutSeconds int `json:"timeoutSeconds,omitempty"`
+
+	// IntervalSeconds defines how often (in seconds) to perform
+	// an execution of status check.
+	// +optional
+	// +kubebuilder:default=10
+	// +kubebuilder:validation:Minimum=1
+	IntervalSeconds int `json:"intervalSeconds,omitempty"`
+
+	// FailureThreshold defines the minimum consecutive failure
+	// for the status check to be considered failed.
+	// +optional
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:Minimum=1
+	FailureThreshold int `json:"failureThreshold,omitempty"`
+
+	// SuccessThreshold defines the minimum consecutive successes
+	// for the status check to be considered successful.
+	// SuccessThreshold only works for `Synchronous` mode.
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	SuccessThreshold int `json:"successThreshold,omitempty"`
+
+	// RecordsHistoryLimit defines the number of record to retain.
+	// +optional
+	// +kubebuilder:default=100
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=1000
+	RecordsHistoryLimit int `json:"recordsHistoryLimit,omitempty"`
+
+	// +optional
+	*EmbedStatusCheck `json:",inline,omitempty"`
+}
+
+type StatusCheckStatus struct {
+	// StartTime represents time when the status check started to execute.
+	// +optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// CompletionTime represents time when the status check was completed.
+	// +optional
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// Count represents the total number of the status check executed.
+	// +optional
+	Count int64 `json:"count,omitempty"`
+
+	// Conditions represents the latest available observations of a StatusCheck's current state.
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []StatusCheckCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// Records contains the history of the execution of StatusCheck.
+	// +optional
+	Records []StatusCheckRecord `json:"records,omitempty"`
+}
+
+type StatusCheckOutcome string
+
+const (
+	StatusCheckOutcomeSuccess StatusCheckOutcome = "Success"
+	StatusCheckOutcomeFailure StatusCheckOutcome = "Failure"
+)
+
+type StatusCheckRecord struct {
+	StartTime *metav1.Time       `json:"startTime"`
+	Outcome   StatusCheckOutcome `json:"outcome"`
+}
+
+type StatusCheckConditionType string
+
+const (
+	// StatusCheckConditionCompleted means the status check is completed.
+	// It will be `True`, in the following scenarios:
+	// 1. the duration is exceeded
```

Comment:
According to this comment, this scenario is duplicated with `DurationExceed`.

Reply:
Yes, one of `DurationExceed`, `FailureThresholdExceed`, `SuccessThresholdExceed` is `True` will also lead `Completed` to be `True`. `Completed` represents whether the status check is completed, and other three conditions represent why the status check completed.

Reply:
Get it.